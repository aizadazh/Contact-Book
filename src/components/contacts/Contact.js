import React, { Component } from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import propTypes from "prop-types";
import { deleteContact } from "../../actions/ContactAction";


export class Contact extends Component {
    state = {
        showContactInfo: false,
    };
    onDeleteClick  = id => {
        this.props.deleteContact(id);
    };
    render(){
        const { id, name, address, phone } = this.props.contact;
        const { showContactInfo} = this.state;
        return(
            <div>
                <h4>
                    {name}{" "}
                    <i 
                        onClick = {() =>
                            this.setState({
                                showContactInfo: !this.state.showContactInfo,
                            })
                        }
                        className = "fas fa-sort-down"
                        style = {{cursor: "pointer"}}
                        />
                    <i 
                        className = "fas fa-times"
                        style = {{cursor: "pointer", float:"right", color: "red"}} 
                        onClick = {this.onDeleteClick.bind(this, id)}
                    />
                    <Link to ={`contact/edit/${id}`}>
                        <i
                            className = "fas fa-pencil-alt"
                            style={{
                                cursor: "pointer",
                                float: "right",
                                color: "grey",                               
                                marginRight: "1rem",
                            }}
                        />
                    </Link>                    
                </h4>
                    {showContactInfo ? (
                        <ul className="list-group">
                            <li className="list-group-item">Address: {address}</li>
                            <li className="list-group-item">Phone: {phone}</li>
                        </ul>
                    ) : null }
            </div>
        );
    }
}
Contact.propTypes = {
    contact: propTypes.object.isRequired,
    deleteContact: propTypes.func.isRequired,
};
export default connect(
    null, 
    { deleteContact },    
)(Contact);
