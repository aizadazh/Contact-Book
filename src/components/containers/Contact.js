import React, { Component } from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import propTypes from "prop-types";
import { deleteContact } from "../../actions/ContactAction";


export class Contact extends Component {
    state = {
        showContactInfo: false,
    };
    onDeleteClick  = id => {
        this.props.deleteContact(id);
    };
    render(){
        const { id, name, address, phone } = this.props.contact;
        console.log(this.props.contact.address.street)
        const { showContactInfo } = this.state;
        let nameOfStreet=''
        if(this.props.contact.address.substring){
            nameOfStreet=address
        }else{
            nameOfStreet=address.street
        }
        return(
            <div>
                <h4>
                    {name}{" "}
                    <i 
                        onClick = {() =>
                            this.setState({
                                showContactInfo: !this.state.showContactInfo,
                            })
                        }
                        className = "fas fa-sort-down"
                        style = {{cursor: "pointer", color: "blue"}}
                        />
                    <i 
                        className = "fas fa-times"
                        style = {{cursor: "pointer", float:"right", color: "red"}} 
                        onClick = {this.onDeleteClick.bind(this, id)}
                    />
                    <Link to ={`contact/edit/${id}`}>
                        <i
                            className = "fas fa-pencil-alt"
                            style={{
                                cursor: "pointer",
                                float: "right",
                                color: "green",                               
                                marginRight: "1rem",
                                fontSize: "0.7em",
                                paddingTop: "0.3em",
                            }}
                        />
                    </Link>                    
                </h4>
                    {showContactInfo ? (
                        <ul className="list-group">
                            <li className="list-group-item">Address: {nameOfStreet}</li>
                            <li className="list-group-item">Phone: {phone}</li>
                        </ul>
                    ) : null }
            </div>
        );
    }
}
Contact.propTypes = {
    contact: propTypes.object.isRequired,
    deleteContact: propTypes.func.isRequired,
};
export default connect(
    null, 
    { deleteContact },    
)(Contact);
