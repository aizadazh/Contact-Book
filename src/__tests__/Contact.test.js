import React from "react";
import { shallow } from "enzyme";
import { Contact } from "../components/containers/Contact";

describe("Testing Contact component", ()=>{
    const props = {
        contact: {
            id: 1,
            name: "Aizada",
            address: "Shopokova",
            phone: "0552440028"
        },
        deleteContact: jest.fn()
    }


describe("Testing availability of classes, tags and icons", ()=>{
    const wrapper = shallow(<Contact {...props} />)
    it('renders correctly', () => {
        expect(wrapper.find('li')).toHaveLength(0);
        expect(wrapper.find('.fas fa-sort-down')).toHaveLength(0);
        expect(wrapper.find('.fas fa-times')).toHaveLength(0);
        expect(wrapper.find('.fas fa-pencil-alt')).toHaveLength(0);
        expect(wrapper.find('h4').text()).toEqual('Aizada <Link />');
        wrapper.setState({showContactInfo: true});

        expect(wrapper.setState({ showContactInfo: true }))

        expect(wrapper).toMatchSnapshot();       
    })
});

});