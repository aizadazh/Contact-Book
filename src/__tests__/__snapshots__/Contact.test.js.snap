// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Contact component Testing availability of classes, tags and icons renders correctly 1`] = `ShallowWrapper {}`;
